scannerlib/configchecks/sql_check.go

```diff
@@ -61,27 +61,31 @@ func (c *MySQLCheck) Exec() (ComplianceMap, error) {
 }
 
 // BenchmarkIDs returns the IDs of the benchmarks associated with this check.
-func (c *MySQLCheck) BenchmarkIDs() []string {
-	// We don't do batching for MySQL checks, so we will always have exactly one ID.
+func (c *SQLCheck) BenchmarkIDs() []string {
+	// We don't do batching for SQL checks, so we will always have exactly one ID.
 	return []string{c.benchmarkID}
 }
 
-func (c *MySQLCheck) String() string {
-	return fmt.Sprintf("[MySQL check with id %q]", c.benchmarkID)
+func (c *SQLCheck) String() string {
+	return fmt.Sprintf("[SQL check with id %q]", c.benchmarkID)
 }
 
 // createSQLChecksFromConfig parses the benchmark config and creates the executable
 // SQL checks that it defines.
-func createSQLChecksFromConfig(ctx context.Context, benchmarks []*benchmark, timeout *timeoutOptions, sq scanapi.SQLQuerier) ([]*MySQLCheck, error) {
+func createSQLChecksFromConfig(ctx context.Context, benchmarks []*benchmark, timeout *timeoutOptions, sq scanapi.SQLQuerier) ([]*SQLCheck, error) {
 	// TODO(b/235991635): Use timeout.
-	checks := []*MySQLCheck{}
+	checks := []*SQLCheck{}
 	for _, b := range benchmarks {
 		for _, alt := range b.alts {
 			for _, sqlCheckInstruction := range alt.proto.GetSqlChecks() {
-				if sqlCheckInstruction.GetTargetDatabase() != ipb.SQLCheck_DB_MYSQL {
-					return nil, errors.New("only MySQL/MariaDB database checks are supported")
+				dbtype, err := sq.SupportedDatabase()
```

Comment:
Are there any tests we can add to sql_check_test.go for this new code? e.g. one where we successfully create a check instruction with a Cassandra target DB, and one where we create one with the wrong type.

Reply:
PR looks good now in general, but please also add some tests for this new logic as mentioned above.

Reply:
Actually I think we can add these in a follow-up PR, so this should be good to go.